File: springbootDynamoDB-sdk-v2/src/main/java/com/dynamodb/service/MusicService.java
package com.dynamodb.service;

import com.dynamodb.dto.Artist;
import com.dynamodb.repositories.MusicRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class MusicService {
	
	@Autowired
	MusicRepository musicRepository;
	
	public Artist findArtistByName(String name) {
		return musicRepository.findArtistByName(name);
	}

}

File: springbootDynamoDB-kotlin-sdk-v2/src/main/kotlin/me/brunosantana/service/MusicService.kt
package me.brunosantana.service

import me.brunosantana.dto.Artist
import me.brunosantana.dto.Song
import me.brunosantana.repositories.MusicRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class MusicService(
	val musicRepository: MusicRepository
) {
	fun findArtistByName(name: String): Artist {
		return musicRepository.findArtistByName(name) ?:
			Artist(
				name = "",
				nationality = "",
				isAwardWinner = false,
				internationalSinger = null,
				actor = 0
			)
	}

	fun saveArtist(artist: Artist){
		musicRepository.saveModel(artist)
	}

	fun saveSong(song: Song){
		musicRepository.saveModel(song)
	}
}
